src/reducers/index.js

```diff
@@ -97,6 +108,21 @@ export default combineReducers({
       }
     }
 
+    if (type === UPDATE_USER_SETTINGS_PENDING) {
+      return {
+        ...state,
+        current: {...state.current, ...meta.params}
+      }
+    }
+
+    if (type === LEAVE_COMMUNITY_PENDING) {
```

Comment:
you need a corresponding `type === LEAVE_COMMUNITY && error` case to undo this, yeah?